pandas/core/indexes/base.py

```diff
@@ -2979,8 +2979,8 @@ def _union(self, other: Index, sort):
                 value_list.extend([x for x in rvals if x not in value_set])
                 return Index(value_list)._values  # do type inference here
 
-        elif not other.is_unique and not self.is_unique:
-            # self and other both have duplicates
+        elif not other.is_unique:
```

Comment:
why doesnt the `not self.is_unique` case go through here?

Reply:
self is our basis in this case and we add things from other, which are not in self, so if self has duplicates and other has not, we are fine, but if other has duplicates which are unique in self, the dups are not added below.

Example:

```
self = Index([0, 0, 1])
other = Index([0, 1, 2])
```

-> we add 2 to lvals and get out result, but if

```
self = Index([0, 1, 2])
other = Index([0, 0, 1])
```

Nothing is missing in this case, so we add nothing to self, but we want to add one 0.

Sorry is tricky to explain, was this clear enough to get an idea why we are doing this? We could in theory send it through I think, but I believe the function is slower than the code below

Reply:
makes sense, thanks